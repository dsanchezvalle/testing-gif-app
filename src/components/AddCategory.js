import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
        //console.log('Handle change activated')
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Handle Submit called', inputValue)
        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
