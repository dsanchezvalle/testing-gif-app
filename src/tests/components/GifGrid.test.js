import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme'
import React from 'react'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GifGrid tests', ()=>{
    test('it should render properly', ()=>{
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const category = 'Goku';
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('it should show items when imgs are loaded through useFetchGifs', ()=>{
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/any.jpg',
                title: 'Any title'
            },
            {
                id: '123',
                url: 'https://localhost/any.jpg',
                title: 'Any title'
            }
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const category = 'Goku';
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})