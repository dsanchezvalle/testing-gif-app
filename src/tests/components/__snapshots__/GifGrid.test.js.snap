// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests it should render properly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid tests it should show items when imgs are loaded through useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Any title"
      url="https://localhost/any.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Any title"
      url="https://localhost/any.jpg"
    />
  </div>
</Fragment>
`;
